### DISCUSSION TITLE
osId having two formats for a single record

### DISCUSSION THREAD

**Discussion by maheshkumargangula** [Sunbird-RC/community#130] [Issues]
@dileepbapat - The value of the `osId` property is having two formats of the data. Below are the details.

1. Invite API response returning `osId` as like `<shardId>-<uuid>`. Ex: `1-1206f8fc-0dd4-43fa-a8fb-f74a3f265cc8`
2. Search API response returning `osId` as like `<uuid>` only. Ex: `1206f8fc-0dd4-43fa-a8fb-f74a3f265cc8`

Read and Update APIs working with both the values. But, If we have multiple `shardId` values for the registry instance. It will be a problem. 

When I updated the configuration of `shardLabel`, the registry service returned a 500 error code.

Cc: @vrayulu

**Comment by pramodkvarma** [comment]
@dileepbapat we also had another thread on on `osId` format. Not sure why we need `shardId` there if the sharding is consistent and using the `osId` as the shard key. Shouldn't the `shardId` be purely internal? 
@tejash-jl

**Reply by dileepbapat** [reply]
This was the interface inherited from opensaber, it appends shard id along with unique id. when the response from search is used it will use the default shard.

However, we can move to virtual slot-based sharding like redis.  but this will break backward compatibility. 

We will take a look at 500 case and update soon.

**Reply by pramodkvarma** [reply]
Oh, OK. Thanks @dileepbapat. Yeah, we can provide it under backward compatibility mode. 

Do update once you have analyzed the details. Thanks.
